pagefaultgames/pokerogue
# [Move] Implement Happy Hour

## What are the changes?

Implements the move [Happy Hour](https://bulbapedia.bulbagarden.net/wiki/Happy_Hour_(move)). Since it's normally an event-only move, nothing gets it naturally currently, even as an egg move.

## Why am I doing these changes?



The move was not yet implemented.

## What did change?


Like how it functions in canon, Happy Hour doubles the prize money from trainers and from money moves like Pay Day and Make it Rain, but does _not_ affect money items like Nuggets. It also expires when the Pokémon are recalled like all other arena tags.

### Screenshots/Videos



https://github.com/pagefaultgames/pokerogue/assets/59298170/eb96877b-2ae1-4d10-9c92-4fe55f4d3f9a


https://github.com/pagefaultgames/pokerogue/assets/59298170/e22bd645-bcfe-401e-b5bd-c0f73dbc6afd


## How to test the changes?




Set the move override to have the move Happy Hour and any other attacking move. Complete a battle with a trainer or having used a money attack with and without Happy Hour in effect, and observe the difference in money earned.

## Checklist
